From 51daddbb1af740355e2bfc5e51b5e80da708dda7 Mon Sep 17 00:00:00 2001
From: Andrew Price <anprice@redhat.com>
Date: Mon, 5 Jan 2015 12:17:03 -0500
Subject: [fs] dlm: fix missing endian conversion of rcom_status flags

Message-id: <20150105121659.GA16383@rhlaptop.andrewprice.me.uk>
Patchwork-id: 102456
O-Subject: [RHEL7.1 PATCH] dlm: fix missing endian conversion of rcom_status flags
Bugzilla: 1175900
RH-Acked-by: Robert S Peterson <rpeterso@redhat.com>
RH-Acked-by: Abhijith Das <adas@redhat.com>
RH-Acked-by: David Teigland <teigland@redhat.com>

Bugzilla: 1175900
Brew: 8420807
Tested: by myself, QE
Upstream:

    commit c07127b48c6367255fb4506e6d6ba6e219205607
    From: Neale Ferguson <neale@sinenomine.net>
    Date: Tue, 14 Oct 2014 15:10:48 -0500
    Subject: [PATCH] dlm: fix missing endian conversion of rcom_status flags

    The flags are already converted to le when being sent,
    but are not being converted back to cpu when received.

    Signed-off-by: Neale Ferguson <neale@sinenomine.net>
    Signed-off-by: David Teigland <teigland@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/dlm/rcom.c b/fs/dlm/rcom.c
index 9d61947..f3f5e72 100644
--- a/fs/dlm/rcom.c
+++ b/fs/dlm/rcom.c
@@ -206,7 +206,7 @@ static void receive_rcom_status(struct dlm_ls *ls, struct dlm_rcom *rc_in)
 
  rs = (struct rcom_status *)rc_in->rc_buf;
 
- if (!(rs->rs_flags & DLM_RSF_NEED_SLOTS)) {
+ if (!(le32_to_cpu(rs->rs_flags) & DLM_RSF_NEED_SLOTS)) {
   status = dlm_recover_status(ls);
   goto do_create;
  }
-- 
1.7.1